org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/JavaFeature.java

```diff
@@ -65,19 +65,19 @@ public enum JavaFeature {
 			Messages.bind(Messages.record_patterns),
 			new char[][] {},
 			false),
-	UNNAMMED_PATTERNS_AND_VARS(ClassFileConstants.JDK22,
+	UNNAMMED_PATTERNS_AND_VARS(ClassFileConstants.JDK23,
 			Messages.bind(Messages.unnamed_patterns_and_vars),
 			new char[][] {},
 			false),
-	IMPLICIT_CLASSES_AND_INSTANCE_MAIN_METHODS(ClassFileConstants.JDK22,
+	IMPLICIT_CLASSES_AND_INSTANCE_MAIN_METHODS(ClassFileConstants.JDK23,
 			Messages.bind(Messages.implicit_classes_and_instance_main_methods),
 			new char[][] {},
 			true),
-	STRING_TEMPLATES(ClassFileConstants.JDK22,
+	STRING_TEMPLATES(ClassFileConstants.JDK23,
 			Messages.bind(Messages.string_templates),
 			new char[][] {},
 			true),
-	STATEMENTS_BEFORE_SUPER(ClassFileConstants.JDK22,
+	STATEMENTS_BEFORE_SUPER(ClassFileConstants.JDK23,
 			Messages.bind(Messages.statements_before_super),
 			new char[][] {},
 			true),
```

Comment:
I don't think you should change those values. Those features were introduced in 22

Reply:
With and without the change JDT UI behaves the same when using JavaSE-23, unfortunately there is no way to enable preview features on JavaSE-22 if constant for support for Java 23 is there (somehow I believe it's UI limitation in JDT).

Reply:
Supporting preview features for several compiler versions is not a priority IMO